File: 1365_how_many_numbers_are_smaller_than_the_current_number/how_many_numbers_are_smaller_than_the_current_number.py
from typing import List

class Solution:
    def smallerNumbersThanCurrent(self, nums: List[int]) -> List[int]:
        count = [0] * 101

        for num in nums:
            count[num] += 1

        for i in range(1, 101):
            count[i] += count[i - 1]

        result = []
        for num in nums:
            result.append(0 if num == 0 else count[num - 1])

        return result


if __name__ == "__main__":
    nums = [8, 1, 2, 2, 3]
    solution = Solution()
    print(solution.smallerNumbersThanCurrent(nums))

File: 1365_how_many_numbers_are_smaller_than_the_current_number/how_many_numbers_are_smaller_than_the_current_number.go
package main

import "fmt"

type Solution struct{}

func (s *Solution) smallerNumbersThanCurrent(nums []int) []int {
	count := make([]int, 101)

	for _, num := range nums {
		count[num]++
	}

	for i := 1; i < 101; i++ {
		count[i] += count[i-1]
	}

	result := make([]int, len(nums))
	for i, num := range nums {
		if num == 0 {
			result[i] = 0
		} else {
			result[i] = count[num-1]
		}
	}
	return result
}

func main() {
	nums := []int{8, 1, 2, 2, 3}
	s := Solution{}
	fmt.Println(s.smallerNumbersThanCurrent(nums))
}
